### PR TITLE
nitunit: improve progress display

### PR DESCRIPTION
**Pull request by privat**
The results are printed progressively instead of at the end of each test-suite.
The progress bar in therefore displayed below the results.

See it in action: https://asciinema.org/a/17u9jwts0nz5dfpxp8p9aab0f

### CODE REVIEW

**Comment by Morriar** [comment]
So cute, +1

File: src/testing/testing_base.nit [hunk 1]
```diff
@@ -117,20 +117,31 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
+		return not opt_no_color.value and opt_verbose.value <= 0
+	end
+
+	# Clear the line if `has_status` (no-op else)
+	fun clear_status
+	do
+		if has_status then printn "\r\x1B[K"
+	end
+
 	# Shoe the full description of the test-case.
```
Comment (Morriar) [inline src/testing/testing_base.nit]: Didn't see that one in the first PR.

Maybe you can fix the `shoe`?

File: src/testing/testing_base.nit [hunk 2]
```diff
@@ -117,21 +117,32 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
 			end
 		end
 
-		if opt_no_color.value then
+		if not has_status then
```
Comment (jcbrinfo) [inline src/testing/testing_base.nit]: Now, with `nitunit -v --no-color`, I get colors… :confused:

File: src/testing/testing_base.nit [hunk 3]
```diff
@@ -117,21 +117,32 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
-			line += " " + more_message
-		end
 		printn "{line}"
 	end
 
-	# Shoe the full description of the test-case.
+	# Is a status bar printed?
+	#
+	# true if color and non-verbose mode
+	fun has_status: Bool
```
Comment (jcbrinfo) [inline src/testing/testing_base.nit]: I find this name confusing considering that `show_unit_status` shows the progress bar even if `has_status == false`.

Comment (privat) [inline src/testing/testing_base.nit]: yeah. bad names confuse things.
I will rename this `status` to `progress_bar` to distinguish with the `show_unit_status`.

**Comment by jcbrinfo** [comment]
Except for the name of `has_status` (may I suggest something like `show_progress`?), :+1: +1

And please resize your window before recording a screencast: it is annoying to have to scroll to see all the output.

**Comment by privat** [comment]
rerol: rename methods and improve output slightly

**Pull request merged**